### PR TITLE
[NPU] add npu matmul op

### PR DESCRIPTION
**Pull request by ShuZihan**

### CODE REVIEW

**Comment by YanhuiDua** [comment]
`2023-08-02 13:41:27 ======================================================================
2023-08-02 13:41:27 FAIL: test_check_output (test_matmul_op_npu.TestMatmulNPUOp)
2023-08-02 13:41:27 ----------------------------------------------------------------------
2023-08-02 13:41:27 Traceback (most recent call last):
2023-08-02 13:41:27   File "/paddle/backends/npu/build/tests/unittests/test_matmul_op_npu.py", line 104, in test_check_output
2023-08-02 13:41:27     self.check_output_with_place(self.place, atol=1e-3)
2023-08-02 13:41:27   File "/paddle/python/tests/op_test.py", line 2161, in check_output_with_place
2023-08-02 13:41:27     static_checker.check()
2023-08-02 13:41:27   File "/paddle/python/tests/op_test.py", line 1928, in check
2023-08-02 13:41:27     self.compare_outputs_with_expects()
2023-08-02 13:41:27   File "/paddle/python/tests/op_test.py", line 1918, in compare_outputs_with_expects
2023-08-02 13:41:27     self.compare_single_output_with_expect(out_name, expect)
2023-08-02 13:41:27   File "/paddle/python/tests/op_test.py", line 1896, in compare_single_output_with_expect
2023-08-02 13:41:27     self._compare_numpy(name, actual_np, expect_np)
2023-08-02 13:41:27   File "/paddle/python/tests/op_test.py", line 1855, in _compare_numpy
2023-08-02 13:41:27     assert (
2023-08-02 13:41:27 AssertionError: Operator (custom_matmul) : Output (Out) shape mismatch, expect shape is (1,), but actual shape is (0,)
2023-08-02 13:41:27 ----------------------------------------------------------------------`

单测挂了，请看下

File: backends/npu/tools/compile.sh [hunk 1]
```diff
@@ -35,19 +35,23 @@ cat <<EOF
 ========================================
 Configuring cmake in build ...
     -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-Release}
+    -DWITH_KERNELS=${WITH_KERNELS:-ON}
```
Comment (qili93) [inline backends/npu/tools/compile.sh]: WITH_KERNELS 这个开关已经没有了，这里是冗余的，不需要添加

File: backends/npu/tools/compile.sh [hunk 2]
```diff
@@ -35,19 +35,23 @@ cat <<EOF
...
     -DWITH_TESTING=${WITH_TESTING:-ON}
     -DWITH_MKLDNN=${WITH_MKLDNN}
     -DWITH_ARM=${WITH_ARM}
+    -DWITH_ASCEND_TRANSFORMER_ACC=${WITH_ASCEND_TRANSFORMER_ACC:-OFF}
```
Comment (qili93) [inline backends/npu/tools/compile.sh:41]: 这个新增的 WITH_ASCEND_TRANSFORMER_ACC 的开关，需要在 https://github.com/PaddlePaddle/PaddleCustomDevice/blob/develop/backends/npu/CMakeLists.txt#L31 这里添加 option 的说明文字，说明这个开关的作用。

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**